Author: Matus Kral <matuskral@me.com>

  add current mode to modelist (when generating default modelist).
  during tv power on (and incomming plugout/plugin events) can happen
  fail on EDID readout. mxc_hdmi generates default modelist - it is
  finaly again replaced by proper EDID read, but in the transition we
  have default. if current mode active is not in the default mode list,
  our actual (and proper mode) is changed for closest match possible
  (match to any mode from this tmp default modelist).
  in few seconds we get again previous EDID red modelist, but hdmi
  configured with another mode than before.

  by adding current mode we make sure no other mode is switched
  during this transition time. 

  (drawback of this solution is improper (unchanged/previous) mode
  stuck in case we are REALLY switching monitor/TV with completely
  different supported modes than the one before)

diff --git a/drivers/video/mxc/mxc_hdmi.c b/drivers/video/mxc/mxc_hdmi.c
--- a/drivers/video/mxc/mxc_hdmi.c
+++ b/drivers/video/mxc/mxc_hdmi.c
@@ -2077,6 +2077,9 @@
 			fb_add_videomode(mode, &hdmi->fbi->modelist);
 	}
 
+	fb_var_to_videomode(&m, &hdmi->fbi->var);
+	fb_add_videomode(&m, &hdmi->fbi->modelist);
+
 	fb_new_modelist(hdmi->fbi);
 
 	console_unlock();
